modules/tinymce/src/core/main/ts/api/ThemeManager.ts

```diff
@@ -16,7 +16,7 @@ export interface Theme {
   execCommand?: (command: string, ui?: boolean, value?: any) => boolean;
   destroy?: () => void;
   init?: (editor: Editor, url: string) => void;
-  renderUI?: () => RenderResult;
+  renderUI?: () => Promise<RenderResult>;
```

Comment:
This needs to be sync and async to not break the public api but it can just be wrapped in a promise if it's a sync value.

```suggestion
  renderUI?: () => Promise<RenderResult> | RenderResult;
```

- [ ] Add fallback to sync api
- [ ] Add test for this new async and sync api by creating a custom themes in the tests and init with those
- [ ] Add changelog items for API change and load/render order change